faucet/internal/http/faucet.go

```diff
@@ -0,0 +1,74 @@
+package http
+
+import (
+	"errors"
+	"html/template"
+	"net/http"
+	"path"
+
+	"github.com/ethereum/go-ethereum/common"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/consensus-shipyard/calibration/faucet/internal/data"
+	"github.com/consensus-shipyard/calibration/faucet/internal/faucet"
+	"github.com/consensus-shipyard/calibration/faucet/internal/platform/web"
+)
+
+type FaucetWebService struct {
+	log            *logging.ZapEventLogger
+	faucet         *faucet.Service
+	backendAddress string
+}
+
+func NewWebService(log *logging.ZapEventLogger, faucet *faucet.Service, backendAddress string) *FaucetWebService {
+	return &FaucetWebService{
+		log:            log,
+		faucet:         faucet,
+		backendAddress: backendAddress,
+	}
+}
+
+func (h *FaucetWebService) handleFunds(w http.ResponseWriter, r *http.Request) {
+	var req data.FundRequest
+	if err := web.Decode(r, &req); err != nil {
+		web.RespondError(w, http.StatusBadRequest, err)
+		return
+	}
+
+	if req.Address == "" {
+		web.RespondError(w, http.StatusBadRequest, errors.New("empty address"))
+		return
+	}
+
+	h.log.Infof(">>> %s -> {%s}\n", r.RemoteAddr, req.Address)
+
+	targetAddr := common.HexToAddress(req.Address)
```

Comment:
It would be cool if we could accept `f4` addresses as input. We could check here if it is a filecoin f4 address, and then convert it into its underlying ethereum address (this is not blocking, but definitely a nice to have). 

In the future we should also consider supporting funding any other type of address through a Fendermint client.